core/src/components/datetime/utils/parse.ts

```diff
@@ -106,3 +108,20 @@ export const parseDate = (val: string | undefined | null): any | undefined => {
     tzOffset,
   };
 };
+
+export const clampDate = (date: string, min?: string, max?: string): string => {
```

Comment:
By the time `processValue` is called, we have already parsed the `min` and `max` values, so we should be able to pass in the `DatetimeParts` instead of the strings: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/datetime/datetime.tsx#L1111-L1113